<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1115</width>
    <height>796</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1115</width>
    <height>796</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>梯度下降</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(236, 236, 236);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="lbl_img">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>170</y>
      <width>420</width>
      <height>500</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">border-color: rgb(211, 211, 211);</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="group_9_to_16">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>110</y>
      <width>300</width>
      <height>560</height>
     </rect>
    </property>
    <widget class="QPushButton" name="pushButton_16">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>490</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_15">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>420</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_14">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>350</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_11">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>140</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_13">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>280</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_10">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>70</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_9">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_12">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>210</y>
       <width>300</width>
       <height>70</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>400</width>
       <height>100</height>
      </size>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(198, 198, 198);</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string>[Blank Page]</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>80</y>
      <width>351</width>
      <height>591</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="styleSheet">
        <string notr="true">border-bottom-color: rgb(211, 211, 211);</string>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:18pt; font-weight:600;&quot;&gt;评估结果：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTextBrowser" name="textBrowser">
       <property name="styleSheet">
        <string notr="true">border-color: rgb(211, 211, 211);</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>80</y>
      <width>231</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QPushButton" name="btn_train">
       <property name="font">
        <font>
         <pointsize>22</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>运行</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_exit">
       <property name="font">
        <font>
         <pointsize>22</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>退出</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="group_A_to_H">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>300</width>
      <height>100</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="styleSheet">
     <string notr="true">selection-background-color: rgb(255, 255, 0);</string>
    </property>
    <widget class="QPushButton" name="pushButton_A">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="text">
      <string>A</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_B">
     <property name="geometry">
      <rect>
       <x>75</x>
       <y>0</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="text">
      <string>B</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_C">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>0</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="text">
      <string>C</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_D">
     <property name="geometry">
      <rect>
       <x>225</x>
       <y>0</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">selection-color: rgb(255, 255, 0);
selection-background-color: rgb(255, 255, 0);</string>
     </property>
     <property name="text">
      <string>D</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_E">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="text">
      <string>E</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_F">
     <property name="geometry">
      <rect>
       <x>75</x>
       <y>50</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="text">
      <string>F</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_G">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>50</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="text">
      <string>G</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_H">
     <property name="geometry">
      <rect>
       <x>225</x>
       <y>50</y>
       <width>75</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>27</pointsize>
      </font>
     </property>
     <property name="text">
      <string>H</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="btn_openFile">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>10</y>
      <width>120</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="cursor">
     <cursorShape>OpenHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>选择数据集</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="gradient_descent_testsize">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>20</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QTextBrowser" name="result_img_path">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>680</y>
      <width>1115</width>
      <height>80</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>26</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QTextBrowser" name="dataset_name">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>10</y>
      <width>210</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">border-color: rgb(255, 255, 255);</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>20</y>
      <width>140</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;测试集大小：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>25</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>方法：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="method">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>25</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>随机梯度下降</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>批梯度下降</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>小批量梯度下降</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>70</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>学习率：</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="learning_rate">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>65</y>
      <width>70</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>105</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>迭代次数：</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="epochs">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>100</y>
      <width>46</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000000</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>140</y>
      <width>72</width>
      <height>15</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>批量大小：</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="batch_size">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>135</y>
      <width>46</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000000</number>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1115</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
